rust/src/transformations/make_stable_expr/expr_alias/mod.rs

```diff
@@ -0,0 +1,73 @@
+use std::collections::BTreeSet;
+use std::mem::replace;
+
+use polars_plan::dsl::Expr;
+
+use crate::core::{ExprFunction, Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprDomain, OuterMetric};
+use crate::error::*;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that renames a column in a LazyFrame.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The alias expression
+pub fn make_expr_alias<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Alias(input, name) = expr else {
+        return fallible!(MakeTransformation, "expected boolean function expression");
+    };
+
+    let t_prior = input.make_stable(input_domain, input_metric)?;
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    let mut output_domain = middle_domain.clone();
+    let old_name = replace(
+        &mut output_domain.active_series_mut()?.field.name,
+        name.as_ref().into(),
+    );
+
+    output_domain.frame_domain.margins = (output_domain.frame_domain.margins)
+        .into_iter()
+        .filter_map(|(k, v)| {
+            let old_len = k.len();
+            let new_k: BTreeSet<_> = (k.into_iter())
+                .map(|col| {
+                    if col == old_name {
+                        name.to_string()
+                    } else {
+                        col
+                    }
+                })
+                .collect();
+            (new_k.len() == old_len).then_some((new_k, v))
+        })
+        .collect();
```

Comment:
A little more explanation here: A comment at the top, or pull out the internals in a named function. I'm not really sure what's happening with the `len`.